modules/services/pass-secret-service.nix

```diff
@@ -16,33 +15,48 @@ in {
     package = mkPackageOption pkgs "pass-secret-service" { };
 
     storePath = mkOption {
-      type = with types; nullOr str;
-      default = null;
-      defaultText = "~/.password-store";
+      type = types.str;
+      default = "";
+      defaultText = "$HOME/.password-store";
```

Comment:
`defaultText` is for reflecting the option's `default` value. If the software uses its own default when the option is `null` or empty, this should be documented in the `description`, or we should just `default` to it to avoid the indirection.

The actual default is also determined by the [password-store module](https://github.com/nix-community/home-manager/blob/669669fcb403e3137dfe599bbcc26e60502c3543/modules/programs/password-store.nix#L62-L63), so maybe this would be best explained with a `literalDocBook`.

Reply:
Yeah, I never could decide how best to set the default. I want it to be simple where if they don't define it, it uses the application default, but doesn't forcibly couple itself to the `password-store` module if the person doesn't want to either use `password-store` or wants to have a separate store for the `secrets` service. For example, I don't actually use `password-store` currently outside of the secrets service.

Additionally, while `pass` itself uses the `PASSWORD_STORE_DIR` environment variable, `pass-secret-service` will not respect it, as the underlying project only checks the environment for that variable if you use the CLI (see <https://github.com/aviau/python-pass/blob/7b633e0034ad409872d45f09e965ffbe9ebb9700/pypass/passwordstore.py#L61>), and instead defaults to `$HOME/.password-store`. Previous behavior just configured it with the `password-store` module, which is why we added the `mkDefault` choice.

At the same time, it doesn't make sense to define a `null` choice separately from the empty string `""` IMO, since both would indicate "no path given", and the latter might actually cause things to break.

See previous discussion in <https://github.com/nix-community/home-manager/pull/2969#discussion_r877575663>.

Reply:
> At the same time, it doesn't make sense to define a `null` choice separately from the empty string `""` IMO, since both would indicate "no path given", and the latter might actually cause things to break.

What do you mean? `null` is the standard "no value given" value. The fact that the empty string doesn't make sense as a path (hell, sometimes it does) is accidental.

Reply:
> but doesn't forcibly couple itself to the password-store module if the person doesn't want to either use password-store or wants to have a separate store for the secrets service.

It's a default value. Most people will use a single password store, so it's a good default. Those who don't can change it.

Reply:
> `null` is the standard "no value given" value.

Really? There are quite a few other modules that contradict that. A good many modules will use an empty list `[ ]`, an empty attrset `{ }` or the empty string `""` rather than `null`, including ones like `kakoune`, `gpg`, and `bspwm`. Heck, Nix even has an entire challenge around attribute paths that may or may not exist, but those don't resolve to `null` when accessed, they throw an error. I'd argue since `""` likely doesn't work due to the provision of a path flag in this context, `""` makes for an adequate "undefined" value, while making the system simpler.

Reply:
> > but doesn't forcibly couple itself to the password-store module if the person doesn't want to either use password-store or wants to have a separate store for the secrets service.
> 
> It's a default value. Most people will use a single password store, so it's a good default. Those who don't can change it.

So here's another challenge if we do that: you now _have_ to enable the separate password store, in order for the default configuration to be evaluated. I don't use `password-store` for literally anything that isn't the secrets-service, and while I don't consider myself "most people", it seems awfully bold to presume what most people will do in any case. In any case, that's at best a reason to leave the defaults for both the same value separately, not to couple them together -- and "those who don't can change it"

Reply:
> Really?

In Nix, yes. In languages with a better type system, you would express this type as `Maybe String` or `Option String`, and the default value would be `Nothing` or `None`. `null` is a cheap approximation of that.

In types that have merging semantics, like `lines`, `listOf t` or `attrsOf t`, it makes sense for the default value to be the [neutral element](https://en.wikipedia.org/wiki/Identity_element) of the merging operation (`""`, `[]` and `{}` respectively). This isn't the case here since we can't combine two different paths.

Using `""` as a "default" or "unset" string value is common in shells, for lack of an alternative: everything is a string. Nix has at least an ersatz of a decent type system, let's use it.

Reply:
> do that

Wait, do what? I'm not sure what we're talking about; I'm not suggesting any changes to the modules, only to the documentation.

Reply:
Just revert the type as it was.

Reply:
Should be good with 851df4f, but still waiting on CI to make sure